graylog2-server/src/main/java/org/graylog2/rest/resources/streams/StreamResource.java

```diff
@@ -491,14 +480,61 @@ public Response cloneStream(@ApiParam(name = "streamId", required = true) @PathP
                 .collect(Collectors.toSet());
         streamService.addOutputs(savedStreamObjectId, outputIds);
 
-        final Map<String, String> result = ImmutableMap.of("stream_id", savedStreamId);
+        var result = new StreamCreatedResponse(savedStreamId);
         final URI streamUri = getUriBuilderToSelf().path(StreamResource.class)
             .path("{streamId}")
             .build(savedStreamId);
 
         return Response.created(streamUri).entity(result).build();
     }
 
+    @PUT
+    @Path("/indexSet/{indexSetId}")
+    @Timed
+    @ApiOperation(value = "Assign multiple streams to index set")
+    @ApiResponses(value = {
+            @ApiResponse(code = 404, message = "Index set not found.")
+    })
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @AuditEvent(type = AuditEventTypes.STREAM_UPDATE)
+    public Response assignToIndexSet(@ApiParam(name = "indexSetId", required = true) @PathParam("indexSetId") String indexSetId,
+                                    @ApiParam(name = "JSON body", required = true) @Valid @NotNull List<String> streamIds) {
+        checkPermission(RestPermissions.INDEXSETS_READ, indexSetId);
```

Comment:
INDEXSETS_EDIT?

Reply:
No, the index set itself is not manipulated. Also, this would result in different resources than when creating/updating a single stream (which was the default way to assign a stream to an index set before).